Initial LVGL doc for ESPHome (#3678)
OttoWinter/esphomedocs: components/number/lvgl.rst

.. _lvgl-num:

LVGL Number
===========

.. seo::

>     :description: Instructions for setting up a LVGL widget number component.

Reviewer: ```suggestion
    :description: Instructions for setting up an LVGL widget number component.
```

    :image: ../images/lvgl_c_num.png

> The ``lvgl`` number platform creates a number component from a LVGL widget

Reviewer: ```suggestion
The ``lvgl`` number platform creates a number component from an LVGL widget
```

and requires :ref:`LVGL <lvgl-main>` to be configured.

> Supported widgets are :ref:`lvgl-wgt-arc`, :ref:`lvgl-wgt-bar`, :ref:`lvgl-wgt-sli` and :ref:`lvgl-wgt-spb`. A single number supports a single widget, thus you need to choose among which one's state you want to use.

Reviewer: ```suggestion
Supported widgets are :ref:`lvgl-wgt-arc`, :ref:`lvgl-wgt-bar`, :ref:`lvgl-wgt-sli` and :ref:`lvgl-wgt-spb`.
There is a one-to-one relationship between any given widget and its associated number component.
```

Configuration options:
----------------------

- **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.

> - **name** (**Required**, string): The name of the number.

Reviewer: ```suggestion
```
